src/libaktualizr/primary/reportqueue.h

```diff
@@ -76,11 +76,10 @@ class ReportQueue {
   std::shared_ptr<HttpInterface> http;
   std::thread thread_;
   std::condition_variable cv_;
-  std::mutex thread_mutex_;
-  std::mutex queue_mutex_;
+  std::mutex m_;
   std::queue<std::unique_ptr<ReportEvent>> report_queue_;
   Json::Value report_array{Json::arrayValue};
-  bool shutdown_;
+  bool shutdown_{false};
```

Comment:
That's fine but I'd suggest removing it from the initializer list of the class' constructor, so that we keep one source of truth